randutil/rand.go

```diff
@@ -32,34 +34,72 @@ import (
 	"time"
 )
 
-func init() {
-	// golang does not init Seed() itself
-	rand.Seed(time.Now().UnixNano() + int64(os.Getpid()))
+var defaultSeeder = SeedDatePid
+
+// SeedDatePid provides a basic seed value based only on
+// time and os PID value.
+func SeedDatePid() int64 {
+	return time.Now().UnixNano() + int64(os.Getpid())
 }
 
-var moreMixedSeedOnce sync.Once
-
-func moreMixedSeed() {
-	moreMixedSeedOnce.Do(func() {
-		h := sha256.New224()
-		// do this instead of asking for time and pid again
-		var b [8]byte
-		rand.Read(b[:])
-		h.Write(b[:])
-		// mix in the hostname
-		if hostname, err := os.Hostname(); err == nil {
-			h.Write([]byte(hostname))
-		}
-		// mix in net interfaces hw addresses (MACs etc)
-		if ifaces, err := net.Interfaces(); err == nil {
-			for _, iface := range ifaces {
-				h.Write(iface.HardwareAddr)
-			}
+// SeedDatePidHostMac provides a seed value that in addition to
+// time and os PID, also takes into account the device hostname
+// and network inteface MAC addresses. This may be required when
+// you are trying to randomize the time of actions within a
+// fleet of similar devices, where the time and PID values may
+// not provide enough variance within the device pool.
+func SeedDatePidHostMac() int64 {
```

Comment:
Similar naming comment to above, though I'm not sure what we'd call this. Perhaps `AutoSeedWithNetwork`. Or can this functionality just always be in `Autoseed` so the user doesn't have to think about it?

Reply:
Would ```FleetDeviceSeeder``` work for you? This is a specialised seeder that was designed specifically for load balancing between devices in a customer fleet (for Ubuntu Core).

Reply:
Sounds good!